src/k8s/pkg/k8sd/setup/certificates.go

```diff
@@ -2,120 +2,84 @@ package setup
 
 import (
 	"fmt"
+	"io/fs"
 	"os"
 	"path"
 
 	"github.com/canonical/k8s/pkg/k8sd/pki"
 	"github.com/canonical/k8s/pkg/snap"
 )
 
-func EnsureExtDatastorePKI(snap snap.Snap, certificates *pki.ExternalDatastorePKI) error {
-	toWrite := map[string]string{
-		path.Join(snap.EtcdPKIDir(), "ca.crt"):     certificates.DatastoreCACert,
-		path.Join(snap.EtcdPKIDir(), "client.key"): certificates.DatastoreClientKey,
-		path.Join(snap.EtcdPKIDir(), "client.crt"): certificates.DatastoreClientCert,
+// ensureFile creates fname with the specified contents, mode and owner bits.
+// ensureFile will delete the file if contents is an empty string.
```

Comment:
this "side-effect" is never executed, is it? The file path is always set.
As a side-note, I'm not a big fan of this. That behavior is not really obvious from the function signature.

Reply:
we delete the file if `contents` is empty. the `fname` can obviously not be an empty string. or do i not understand your comment?

Reply:
nevermind - my comment made no sense...